File: URI/1013.c
#include<stdio.h>
#include<stdlib.h>
#include<math.h>

	int a,b,c,m1,m2;

void main(){
	scanf("%d %d %d",&a,&b,&c);
	m1=(a+b+abs(a-b))/2;
	m2=(m1+c+abs(m1-c))/2;
	printf("%d eh o maior\n",m2);
}

File: URI/Haskell/1013.hs
main = do
    l <- getLine
    let lista = words l
    putStrLn $ show (maior (maior (read $ head lista) (read $ lista !! 1)) (read $ lista !! 2)) ++ " eh o maior"

maior :: Int -> Int -> Int
maior a b = (a + b + abs (a-b)) `div` 2